scripts/mdx-transforms/exclude-terraform-content/index.test.mjs

```diff
@@ -0,0 +1,203 @@
+ */
+
+import { describe, it, expect, vi } from 'vitest'
+import { transformExcludeTerraformContent } from './index.mjs'
+import remark from 'remark'
+import remarkMdx from 'remark-mdx'
+
+const runTransform = async (markdown, filePath) => {
+	const processor = await remark()
+		.use(remarkMdx)
+		.use(transformExcludeTerraformContent, {
+			filePath,
+		})
+		.process(markdown)
+	return processor.contents
+}
+
+const ptfeFilePath = 'ptfe-releases/some-file.md'
+
+describe('transformExcludeTerraformContent', () => {
+	it('should remove content within TFC:only blocks', async () => {
+		const markdown = `
+<!-- BEGIN: TFC:only -->
+This content should be removed.
+<!-- END: TFC:only -->
+This content should stay.
+`
+		const result = await runTransform(markdown, ptfeFilePath)
+
+		expect(result).toBe('This content should stay.\n')
+	})
+
+	it('should throw an error for mismatched block names', async () => {
+		const mockConsole = vi.spyOn(console, 'error').mockImplementation(() => {})
+		const markdown = `
+<!-- BEGIN: TFC:only -->
+This content should be removed.
+<!-- END: TFC:other -->
+`
+		await expect(async () => {
+			return await runTransform(markdown, ptfeFilePath)
+		}).rejects.toThrow('Mismatched block names')
+		expect(mockConsole).toHaveBeenCalledOnce()
+	})
+
+	it('should throw an error for unexpected END block', async () => {
+		const markdown = `
+<!-- END: TFC:only -->
+`
+		await expect(async () => {
+			return await runTransform(markdown, ptfeFilePath)
+		}).rejects.toThrow('Unexpected END block')
+	})
+
+	it('should throw an error for unexpected BEGIN block', async () => {
+		const markdown = `
+<!-- BEGIN: TFC:only -->
+<!-- BEGIN: TFC:only -->
+`
+		await expect(async () => {
+			return await runTransform(markdown, ptfeFilePath)
+		}).rejects.toThrow('Unexpected BEGIN block')
+	})
+
+	it('should throw an error if no block could be parsed from BEGIN comment', async () => {
+		const markdown = `
+<!-- BEGIN:  -->
+This content should be removed.
+<!-- END: TFC:only -->
+`
+		await expect(async () => {
+			return await runTransform(markdown, ptfeFilePath)
+		}).rejects.toThrow('No block could be parsed from BEGIN comment')
+	})
+
+	it('should throw an error if no block could be parsed from END comment', async () => {
+		const markdown = `
+<!-- BEGIN: TFC:only -->
+This content should be removed.
+<!-- END:  -->
+`
+		await expect(async () => {
+			return await runTransform(markdown, ptfeFilePath)
+		}).rejects.toThrow('No block could be parsed from END comment')
+	})
+
+	it('should throw an error for blocks that do not match directive', async () => {
+		const markdown = `
+<!-- BEGIN: TFE:only -->
+This content should be removed.
+<!-- END: TFE:only -->
+`
+
+		await expect(async () => {
+			return await runTransform(markdown, ptfeFilePath)
+		}).rejects.toThrow('Directive could not be parsed')
+	})
+
+	it('should remove TFC:only content and leave TFEnterprise:only content for ptfe-releases', async () => {
```

Comment:
Love that you wrote tests for them together. ❤️

Reply:
No problem! Can you imagine if they didn't work together 😂